README.md

```diff
@@ -26,121 +26,217 @@ npm install -g w3up
+
+Options:
+  --version  Show version number                                       [boolean]
+  --help     Show help                                                 [boolean]
+
+Examples:
+  w3up id                                 ID loaded: did:key:z6MkiWm...
+  w3up whoami                             DID:12345...
+  w3up list                               bafy...
+                                          bafy...
+  w3up upload ../../duck.png              uploaded bafy...
+  w3up remove bafy...                     unlinked bafy...
+  w3up car-to-dot ../duck.car             generated examination file
+  w3up generate-car ../duck.png duck.car  CAR created ../duck.png => duck.ca
+```
 
-#### `whoami`
+
+### Registration
+
+Most of the commands in `w3up` require a registered identity keypair.
+
+Registration is a two-step process:
+
+1. [`w3up id`](#id) generates a new identity keypair.
+2. [`w3up register`](#register) associates the identity key with your email address.
+
+Once registered, you can verify that your identity has been registered using [`w3up whoami`](#whoami), which displays the id for your w3up account.
+
+#### `id`
+
+> Displays your identity keypair, creating it if it does not yet exist.
 
 ```sh
-w3up whoami
+w3up id
 ```
 
+On first run, generates the public and private key pairs necessary to work with the underlying [UCAN ](https://ucan.xyz/) authorization system.
+
+Running `w3up id` a second time will load your key from disk instead of generating a new one.
+
+You can validate you are registered!
+
 If you have generated the id properly, you'll see your `did:key` printed in the command line. It should look like `did:bafy....`.
 
 #### `register`
 
+> Registers your identity with the w3up service.
+
 ```sh
 w3up register you@example.com
 ```
 
-Register your identity.
-After you've generated an identity, you need to register it with the w3up service. You'll be sent an email with a code you paste into the command line when prompted. If the code matches what the service expects, you're fully registered and can use all the other commands.
+After you've generated an identity, you need to register it with the w3up service. The `w3up register` command will display a message asking you to check your email. Once you click the activation link in the email, the `w3up register` command will complete the registration process and show a success message. 
 
 _Note: Remember to check your spam folder if you suspect you never got the email_
 
-#### General Usage
+#### `whoami`
+
+> Displays the identifier for your w3up account.
 
-After creating your identity and registering with w3up, you should be able to start using the service. The main commands you'll use are
+```sh
+w3up whoami
+```
 
-1. `w3up upload`
-2. `w3up list`
-3. `w3up unlink`
+The `whoami` command displays an identifier for your w3up account. Note that this may differ from your identity key in cases where you have registered multiple identities to the same account.
```

Comment:
> Note that this may differ from your identity key in cases where you have registered multiple identities to the same account

I'm not 100% sure about this... maybe the `whoami` key is always different from the identity key?

Reply:
The whoami key will always be different from the identity key, as it is the "account" key that is registered in access that ties multiple identity keys together (along with email) or other forms of access.